mutation_partition_v2.cc

```diff
@@ -154,80 +153,31 @@ mutation_partition_v2::mutation_partition_v2(const schema& s, const mutation_par
     _rows.clone_from(x._rows, cloner, current_deleter<rows_entry>());
 }
 
-mutation_partition_v2::mutation_partition_v2(const mutation_partition_v2& x, const schema& schema,
-        query::clustering_key_filter_ranges ck_ranges)
-        : _tombstone(x._tombstone)
-        , _static_row(schema, column_kind::static_column, x._static_row)
-        , _static_row_continuous(x._static_row_continuous)
-        , _rows()
-        , _row_tombstones(x._row_tombstones, range_tombstone_list::copy_comparator_only())
+mutation_partition_v2::mutation_partition_v2(const schema& s, mutation_partition&& x)
+    : _tombstone(x.partition_tombstone())
+    , _static_row(std::move(x.static_row()))
+    , _static_row_continuous(x.static_row_continuous())
+    , _rows(std::move(x.mutable_clustered_rows()))
 #ifdef SEASTAR_DEBUG
-        , _schema_version(schema.version())
+    , _schema_version(s.version())
 #endif
 {
-#ifdef SEASTAR_DEBUG
-    assert(x._schema_version == _schema_version);
-#endif
-    try {
-        for(auto&& r : ck_ranges) {
-            for (const rows_entry& e : x.range(schema, r)) {
-                auto ce = alloc_strategy_unique_ptr<rows_entry>(current_allocator().construct<rows_entry>(schema, e));
-                _rows.insert_before_hint(_rows.end(), std::move(ce), rows_entry::tri_compare(schema));
-            }
-            for (auto&& rt : x._row_tombstones.slice(schema, r)) {
-                _row_tombstones.apply(schema, rt.tombstone());
-            }
+    auto&& tombstones = x.mutable_row_tombstones();
```

Comment:
mutable_row_tombstones is very confusingly named. Let's rename it later.

Reply:
Well, it will be deleted later, I presume.

Reply:
Why not in this patch?

Reply:
I'm confused. It is removed in this patch, yet here it is.

Reply:
`mutable_row_tombstones()` is a preexisting thing which belongs to the old `mutation_partition`, 'x' here is a `mutation_partition` object.

Reply:
We drop `mutable_row_tombstones()` from mutation_partition_v2, not from `mutation_partition`.

Reply:
Ahh, it's a cross-mutation-partition-version constructor.